planner/core/stats.go

```diff
@@ -254,12 +287,31 @@ func (ds *DataSource) generateIndexMergeOrPaths() {
 	}
 }
 
+//
+func (ds *DataSource) isInIndexMergeHints(name string) bool {
+	if ds.indexMergeHints == nil ||
+		(len(ds.indexMergeHints) == 1 && ds.indexMergeHints[0].IndexNames == nil) {
+		return true
+	}
+	for _, hint := range ds.indexMergeHints {
+		for _, index := range hint.IndexNames {
+			if name == index.L {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 // accessPathsForConds generates all possible index paths for conditions.
 func (ds *DataSource) accessPathsForConds(conditions []expression.Expression, usedIndexCount int) []*accessPath {
 	var results = make([]*accessPath, 0, usedIndexCount)
 	for i := 0; i < usedIndexCount; i++ {
 		path := &accessPath{}
 		if ds.possibleAccessPaths[i].isTablePath {
+			if !(ds.isInIndexMergeHints("pk") || ds.isInIndexMergeHints("PK")) {
```

Comment:
Since you are checking with `index.L`, it is useless to pass `PK`. And it should be `primary` rather than `PK`.